delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/StoragesStatistics.scala

```diff
@@ -1,49 +1,107 @@
 import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment
 import ch.epfl.bluebrain.nexus.delta.sourcing.model.ProjectRef
-import monix.bio.{IO, UIO}
+import io.circe.literal._
+import io.circe.{DecodingFailure, JsonObject}
+import monix.bio.IO
 
 trait StoragesStatistics {
 
   /**
-    * Retrieve the current statistics for all storages
-    */
-  def get(): UIO[StorageStatsCollection]
-
-  /**
-    * Retrieve the current statistics for storages in the given project
-    */
-  def get(project: ProjectRef): UIO[Map[Iri, StorageStatEntry]]
-
-  /**
-    * Retrieve the current statistics for the given project
+    * Retrieve the current statistics for a given storage in the given project
     */
   def get(idSegment: IdSegment, project: ProjectRef): IO[StorageFetchRejection, StorageStatEntry]
 
 }
 
-// TODO implements storage statistics
 object StoragesStatistics {
 
-  def apply(): StoragesStatistics = new StoragesStatistics {
-
-    override def get(): UIO[StorageStatsCollection] = UIO.pure(StorageStatsCollection.empty)
-
-    /**
-      * Retrieve the current statistics for storages in the given project
-      */
-    override def get(project: ProjectRef): UIO[Map[Iri, StorageStatEntry]] = UIO.pure(Map.empty)
+  /**
+    * @param client
+    *   the Elasticsearch client
+    * @param fetchStorageId
+    *   the function to fetch the storage ID
+    * @param indexPrefix
+    *   the index prefix
+    * @return
+    *   StorageStatistics instance
+    */
+  def apply(
+      client: ElasticSearchClient,
+      fetchStorageId: (IdSegment, ProjectRef) => IO[StorageFetchRejection, Iri],
+      indexPrefix: String
+  ): StoragesStatistics = {
+    val search = (jsonObject: JsonObject) =>
+      client.search(jsonObject, Set(eventMetricsIndex(indexPrefix).value), Query.Empty)()
 
-    /**
-      * Retrieve the current statistics for the given project
-      */
-    override def get(idSegment: IdSegment, project: ProjectRef): IO[StorageFetchRejection, StorageStatEntry] =
-      IO.raiseError(StorageNotFound(Iri.unsafe(""), project))
+    (idSegment: IdSegment, project: ProjectRef) => {
+      for {
+        storageId <- fetchStorageId(idSegment, project)
+        query     <- storagesStatisticsQuery(project, storageId).hideErrors
+        result    <- search(query).hideErrors
+        stats     <- IO.fromEither(result.as[StorageStatEntry]).hideErrors
+      } yield stats
+    }
   }
 
+  /**
+    * @param projectRef
+    *   the project on which the statistics should be computed
+    * @param storageId
+    *   the ID of the storage on which the statistics should be computed
+    * @return
+    *   a query for the total number of files and the total size of a storage in a given project
+    */
+  private def storagesStatisticsQuery(projectRef: ProjectRef, storageId: Iri): IO[DecodingFailure, JsonObject] =
+    IO.fromEither {
+      json"""
+     {
+      "query": {
+        "bool": {
+          "filter": [
+            {
```

Comment:
I think we can opt here to a different formatting than in the files to have it a little more compact where for example we put  all the term part as one single line:
```json
{ "term": {"project": $projectRef } }
```

Reply:
Looks better now